### DISCUSSION TITLE
Prisma cleared all database rows automatic without any reason

### DISCUSSION THREAD

**Discussion by Ameriq8** [prisma/prisma#16703] [Q&A] [answered]
Prisma cleared all database rows automatic without any reason.
OS: Linux/Windows
Lang: Typescript
Prisma Client Version: 4.7.1

**Comment by janpio** [comment, marked as answer]
That is pretty much impossible. There is no functionality in Prisma ORM that could do that unless you explicitly instruct it.

But things happen of course, so let's figure this out.
Was this after using Prisma CLI on the terminal or Prisma Client in your application?

**Reply by Ameriq8** [reply]
This is the third time it happens with me.

**Reply by Ameriq8** [reply]
After I create new model it happened

**Reply by janpio** [reply]
What does that mean? What code do you execute? How do you figure out afterwards that all the rows are "cleared"?

**Reply by Ameriq8** [reply]
I don't know when it happens. I got the problem after when I try to start debugging my code for ex. the time it happend after send a request to my create payment endpoint to create new session and I got res from the api told me I don't have auth token. Becuase I storage all users sessions in database

**Reply by janpio** [reply]
Please be more precise. You said you "create new model". What kind of model? What function are you using to do that? No other calls to Prisma or usage of the CLI at that time that could influence this?

When you call a `create` method, there is no code that could _delete_ other data. The only way to remove some data is to use a `delete` method, execute a CLI command that prompts you to reset the database and then you choose "Yes" explicitly or run some raw query that explicitly deletes data.

**Reply by Ameriq8** [reply]
Do u want a copy from my schema file to see it.

**Reply by Ameriq8** [reply]
When "I said create new model", I mean add it to schema file

**Reply by Ameriq8** [reply]
Oh my god the problem comes from `migrate dev` command I didn't read everything when I use the command. it warn me.
Okay bro thank u so much for ur help

**Reply by janpio** [reply]
Phew, you had me worried for a bit.

I assume you added a model to your schema file, then ran `migrate dev` and it either threw a warning that you accepted which indeed reset the database, or it created some migration SQL that also warned that it might lead to data loss?

**Reply by Ameriq8** [reply]
Yes bro thank u for ur time.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
